### PR TITLE
Implements fs.readdir and readdirSync

### PR DESCRIPTION
**Pull request by glistening**
Currently, fs.readdir and readdirSync do not work on nuttx.
Since nuttx does not provides scandir API, it will be implemented
using opendir, readdir, closedir.

In addition, TCs for fs.unlink, rename, writefile are included in TC lists.
(They had been abandoned after changing test driver from python to js.)

IoT.js-DCO-1.0-Signed-off-by: Sanggyu Lee sg5.lee@samsung.com

### CODE REVIEW

**Comment by glistening** [comment]
This patch relates to #28.

File: src/module/iotjs_module_fs.cpp [hunk 1]
```diff
@@ -62,6 +62,20 @@ static void After(uv_fs_t* req) {
...
+        int r;
+        uv_dirent_t ent;
+        uint32_t idx = 0;
+        /* make an empty javascript object */
+        JObject ret(0, (const char*)NULL);
+        while ((r = uv_fs_scandir_next(req, &ent)) != UV_EOF) {
+          ret.SetPropertyByIdx(idx, JObject(ent.name));
+          idx++;
+        }
+        jarg.Add(ret);
```
Comment (zherczeg) [inline src/module/iotjs_module_fs.cpp]: Does this compatible with the C jarg implementation?

Comment (glistening) [inline src/module/iotjs_module_fs.cpp]: No. I assume you are talking about #485. All current implementation is based on C++ implementaion (except for String). IMHO, I think switching to C needs more time for discussing and shaping the APIs.

Comment (zherczeg) [inline src/module/iotjs_module_fs.cpp]: Perhaps we should wait for landing that first to avoid too much maintenace.

Comment (glistening) [inline src/module/iotjs_module_fs.cpp]: Do you mean stop all development before completing C switching?

Comment (zherczeg) [inline src/module/iotjs_module_fs.cpp]: Not all. But I hope that patch big lands very soon.

**Comment by yichoi** [comment]
@glistening please rebase to the master

File: src/module/iotjs_module_fs.cpp (hunk 1 above)
Comment (glistening) [inline src/module/iotjs_module_fs.cpp]: Finally #485 landed.  I updated my patch.

**Comment by glistening** [comment]
About 3 hours ago, I had updated to the master. Meanwhile, my another commit is merged. Should I rebase again?

**Comment by glistening** [comment]
I rebased again.

File: test/run_pass/test_fs_readdir.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  res += items[i] + '\n';
+assert.equal(res, ans);
+
+res = "";
+fs.readdir(path, function(err, items) {
+  assert.equal(err, null);
+  items.sort();
+  for (i = 0; i < items.length; i++) {
+    res += items[i] + '\n';
+  }
```
Comment (zherczeg) [inline test/run_pass/test_fs_readdir.js]: What type of for is preferred? Is it intentional to use two different styles?

Comment (glistening) [inline test/run_pass/test_fs_readdir.js]: I am sorry. I placed console.log in the brace-enclosed block for debug. I did not remove braces by mistake. I will fix it. Thank you for review.

**Comment by LaszloLango** [comment]
LGTM

**Pull request merged**